tokio/src/sync/watch.rs

```diff
@@ -10,24 +10,64 @@
 //!
 //! [`channel`] returns a [`Sender`] / [`Receiver`] pair. These are the producer
 //! and consumer halves of the channel. The channel is created with an initial
-//! value. The **latest** value stored in the channel is accessed with
-//! [`Receiver::borrow()`]. Awaiting [`Receiver::changed()`] waits for a new
-//! value to be sent by the [`Sender`] half.
+//! value.
+//!
+//! Each [`Receiver`] independently tracks the last value *seen* by its caller.
+//!
+//! To access the **latest** value stored in the channel and mark it as *seen*,
+//! use [`Receiver::borrow_and_update()`].
+//!
+//! To access the latest value **without** changing its state to *seen*, use
+//! [`Receiver::borrow()`]. (If the value has already been marked *seen*,
+//! [`Receiver::borrow()`] is equivalent to [`Receiver::borrow_and_update()`].)
+//!
+//! ## Change notifications
+//!
+//! The [`Receiver`] half provides an asynchronous [`changed`] method. This
+//! method is ready when a new, *unseen* value is sent via the [`Sender`] half.
+//!
+//! * The [`changed`] method returns `Ok(())` on receiving a new value, or
+//!   `Err(_)` if the [`Sender`] has been closed.
+//! * On completion, the [`changed`] method marks the new value as *seen*. If
+//!   [`Receiver::changed()`] is called again, it will not return immediately
+//!   unless a subsequent value is sent.
```

Comment:
Perhaps it would make sense to add this point somewhere here?

 * If the latest value is unseen when calling `changed`, then `changed` will return immediately. If the latest message is seen, then it will sleep until a new message is sent.

I don't think the current way its written make this sufficiently clear.

Then the second point can also be simplified to this:

 * On completion, the `changed` method marks the new value as *seen*.

Reply:
I like this a lot. Updated with this + also a mention that the sender can be dropped.